src/main.rs

```diff
@@ -50,6 +49,26 @@ macro_rules! plural {
     };
 }
 
+fn is_rept(node: &Node) -> bool {
+    matches!(node.type_data(), NodeType::Rept(..))
+}
+
+fn rept_parent_non_rept<'a>(object: &'a Object, node: &'a Node) -> Option<&'a Node> {
+    if is_rept(node) {
```

Comment:
Consider `.then(|| { ...`?

Reply:
Not sure if that's necessary after the other edits.